flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobExceptionsHandler.java

```diff
@@ -49,15 +50,15 @@
 /**
  * Handler serving the job exceptions.
  */
-public class JobExceptionsHandler extends AbstractExecutionGraphHandler<JobExceptionsInfo, JobMessageParameters> implements JsonArchivist {
+public class JobExceptionsHandler extends AbstractExecutionGraphHandler<JobExceptionsInfo, JobExceptionsMessageParameters> implements JsonArchivist {
 
-	static final int MAX_NUMBER_EXCEPTION_TO_REPORT = 20;
+	private static int maxNumberExceptionToReport = 20;
```

Comment:
it's a bad idea to use a mutable field for passing arguments around. Revert this change and add another argument to `createJobExceptionsInfo()` instead and propagate it to `handleRequest()`.